docs/tutorials.rst

```diff
@@ -0,0 +1,178 @@
+        error.__cause__ = exc
+        emit.error(error)
+    except Exception as exc:
+        error = CraftError(f"Application internal error: {exc!r}") 
+        error.__cause__ = exc
+        emit.error(error)
+    else:
+        emit.ended_ok()
+
+Finally, put both chunks of code in a ``example-app.py`` file, and (having the virtualenv you prepared at the beginning still activated), run it. You should see the help message for the whole application (as a command is missing, which would be the same output if you pass the `help`, `-h` or `--help` parameters)::
+
+    $ python example-app.py
+    Usage:
+        example-app [help] <command>
+    
+    Summary:    Example application for the craft-cli tutorial.
+    
+    Global options:
+           -h, --help:  Show this help message and exit
+        -v, --verbose:  Show debug information and be more verbose
+          -q, --quiet:  Only show warnings and errors, not progress
+          -t, --trace:  Show all information needed to trace internal
+                        behaviour
+    
+    Starter commands:
+    
+    Commands can be classified as follows:
+              Example:  unlink
+    
+    For more information about a command, run 'example-app help <command>'.
+    For a summary of all commands, run 'example-app help --all'.
+
+Ask help for specifically for the command::
+
+    $ python example-app.py help unlink
+    Usage:
+        example-app unlink [options] <filepath>
+    
+    Summary:
+        Remove the indicated file.
+    
+        A file needs to be indicated. It is an argument error if the path does not exist
+        or it's a directory.
+    
+        It will return succesfully if the file was properly removed.
+    
+    Options:
+           -h, --help:  Show this help message and exit
+        -v, --verbose:  Show debug information and be more verbose
+          -q, --quiet:  Only show warnings and errors, not progress
+          -t, --trace:  Show all information needed to trace internal
+                        behaviour
+    
+    For a summary of all commands, run 'example-app help --all'.
+    
+Time to run the command on a file, you should see the successful message::
+
+    12:00:03|facundo@camdeonita:~/canonical/craft-cli (improve-readme *)$ touch testfile
+    12:02:29|facundo@camdeonita:~/canonical/craft-cli (improve-readme *)$ ls testfile2
+    ls: no se puede acceder a 'testfile2': No existe el archivo o el directorio
+    12:02:37|facundo@camdeonita:~/canonical/craft-cli (improve-readme *)$ ls testfile
+    testfile
+    12:02:39|facundo@camdeonita:~/canonical/craft-cli (improve-readme *)$ env/bin/python example-app.py unlink testfile 
+    File removed succesfully.                                                                                                                                                                            
+    12:02:52|facundo@camdeonita:~/canonical/craft-cli (improve-readme *)$ LANG=C ls testfile
```

Comment:
And in this command the locale setting leaked to the tutorial :)

Reply:
yes, the whole block needs processing, updated.